endpoints/openrtb2/auction.go

```diff
@@ -354,12 +374,20 @@ func sendAuctionResponse(
 	account *config.Account,
 	labels metrics.Labels,
 	ao analytics.AuctionObject,
+	seatNonBid *openrtb_ext.NonBidCollection,
 ) (metrics.Labels, analytics.AuctionObject) {
 	hookExecutor.ExecuteAuctionResponseStage(response)
 
+	stageOutcomes := hookExecutor.GetOutcomes()
+	seatNonBid.Append(getNonBidsFromStageOutcomes(stageOutcomes))
+	ao.SeatNonBid = seatNonBid.Get()
```

Comment:
This code is also duplicating logic in `defer`. Func `sendAuctionResponse` executes right before the defer block. Do we need it here?

Reply:
In `sendAuctionResponse` , we will always get the non-nil `ao.Response`.
In defer block, we are collecting nonBids from stageOutcomes  only when `ao.Response` is nil.
We need it here, because we need to add the seatNonBids in `resp.ext.prebid.seatnonbid`